src/domain/Processors/VehicleProcessor.ts

```diff
@@ -82,7 +83,12 @@ export abstract class VehicleProcessor<T extends Vehicle> {
         primaryVrm,
         enums.SEARCHCRITERIA.VRM
       );
-      if (primaryVrmRecords.length > 0) {
+      
+      const allTechRecords = primaryVrmRecords
+        .map(wrapper => wrapper.techRecord)
+        .reduce((accumulator, value) => accumulator.concat(value), []);
```

Comment:
```suggestion
      const allTechRecords = []
      primaryVrmRecords
        .map(wrapper => wrapper.techRecord)
        .forEach(techRecord => allTechRecord.push(...techRecord))
```

In general I think of the `reduce` + `concat` pattern as a bit of a no-no, especially since it can easily be avoided.

`push` is much faster than `concat` (mostly because `concat` creates a new array as opposed to `push` which just uses the original array).

Reply:
Although I realise that was @tahayazici 's suggestion now 😆

Reply:
As discussed yesterday, speed is irrelevant here because the number of iterations will never exceed 2 or 3 digits.

Code maintainability should be the no 1 priority, followed by readability. If speed is a concern, the client and then our tech/dev lead will specifically ask for it.

Reply:
I think this is a leftover from me trying to optimise my advent of code solutions 😆